AsyncDisplayKit/ASContextTransitioning.h

```diff
@@ -0,0 +1,36 @@
+//
+//  ASContextTransitioning.h
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 2/4/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import <AsyncDisplayKit/ASDisplayNode.h>
+
+@protocol ASContextTransitioning <NSObject>
+
+/**
+ @abstreact Defines if the given transition is animated
+ */
+- (BOOL)isAnimated;
+
+/**
+ @abstract The frame for the given node before the transition began.
+ @discussion Returns CGRectNull if the node was not in the hierarchy before the transition.
```

Comment:
Could you include in this file a description of why this is a protocol - should the client ever implement objects that conform to it?  Initially it seemed like they would, maybe a controller object managing the transition at a data level that provided frames for transitions near the cellNodeFor...: method.  But this documentation made me think the user would be calling it on a framework-provided object only, unless we are asking clients here to ensure they do return CGRectNull when appropriate?